modules/olm-deprecations-schema.adoc

```diff
@@ -0,0 +1,86 @@
+// Module included in the following assemblies:
+//
+// * operators/understanding/olm-packaging-format.adoc
+
+[id="olm-deprecations-schema_{context}"]
+= olm.deprecations schema
+
+The optional `olm.deprecations` schema defines deprecation information for packages, bundles, and channels in a catalog. Operator authors can use this schema to provide relevant messages about their Operators, such as support status and recommended upgrade paths, to users running those Operators from a catalog.
+
+An `olm.deprecations` schema entry can contain at least one of the following `reference` types, which indicates the deprecation scope. After the Operator is installed, any deprecation messages can be viewed as status conditions on the related `Subscription` object.
+
+.Deprecation `reference` types
+[cols="1,2,1",options="header"]
+|===
+|Type |Scope |Status condition
+
+|`olm.package`
+|Represents the entire package
+|`PackageDeprecated`
+
+|`olm.channel`
+|Represents one channel
+|`ChannelDeprecated`
+
+|`olm.bundle`
+|Represents one bundle version
+|`BundleDeprecated`
+
+|===
+
+Each `reference` type has their own requirements, as detailed in the following example.
+
+[WARNING]
+====
+If an `olm.deprecations` schema is determined invalid by the `opm validate` command, the entire catalog is invalid.
+====
+
+.Example `olm.deprecations` schema with each `reference` type
+[%collapsible]
+====
+[source,yaml]
+----
+schema: olm.deprecations <1>
+package: my-operator <2>
+entries:
+  - reference:
+      schema: olm.package <3>
+    message: | <4>
+    The 'my-operator' package is end of life. Please use the
+    'my-operator-new' package for support.
+  - reference:
+      schema: olm.channel
+      name: alpha <5>
+    message: |
+    The 'alpha' channel is no longer supported. Please switch to the
+    'stable' channel.
+  - reference:
+      schema: olm.bundle
+      name: my-operator.v1.68.0 <6>
+    message: |
+    my-operator.v1.68.0 is deprecated. Uninstall my-operator.v1.68.0 and
+    install my-operator.v1.72.0 for support.
+----
+<1> There can be only one `olm.deprecations` schema per package.
+<2> The `package` name must be present in the bundle and unique across the catalog. There must not be an associated `name` field.
```

Comment:
<1> and <2> are really the same thing.  The requirement is that each deprecation schema `must` have a package reference (<2>) and that must package reference be unique in the catalog (i.e. must be comprehensive w.r.t. all deprecation activities relevant to that package).  

For e.g. the following is valid:
```yaml
schema: olm.deprecations
package: foo
entries:
- reference
    schema: olm.channel
    name: alpha
  message: hey, bad channel, do <beta>
---
schema: olm.deprecations
package: bar
entries:
- reference: 
    schema: olm.bundle
    name: my-operator.v1.68.0
  message: dead version!  go back!
```
because there are two `olm.deprecations` in the catalog, but they relate to discrete packages.